### PR TITLE
Contacts page test

### PR DESCRIPTION
**Pull request by koropalov**

### CODE REVIEW

**Review by vasylshpak — COMMENTED** [review]
Great job!
Can you provide some description of this task?

image

File: cypress.json [hunk 1]
```diff
@@ -3,5 +3,11 @@
   "viewportHeight": 1080,
   "viewportWidth": 1920,
   "experimentalNetworkStubbing": true,
-  "chromeWebSecurity": false
+  "chromeWebSecurity": false,
+
+  "env": {
```
Comment (moran711) [inline cypress.json:8]: Please remove this envs.

File: cypress/integration/contact/contact.test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  });
+
+  it('User list row should have all types of necessary information about the user', () => {
+    cy.get('@table').eq(0).invoke('text').should('match', telephoneNumber);
+    cy.get('@table').eq(1).invoke('text').should('match', email);
+    cy.get('@table').eq(2).invoke('text').should('match', textString);
+  });
+
+  it('Data from the server should be equal with the incoming data', () => {
+    cy.visit(`contacts/${id}`);
```
Comment (moran711) [inline cypress/integration/contact/contact.test.js:33]: Why are you using visit when you can mock responses and use  `cy.get(selector).click()`

File: cypress/integration/contact/contact.test.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/// <reference types="cypress" />
```
Comment (moran711) [inline cypress/integration/contact/contact.test.js:1]: - Why you don't mock responses from server? We don't have any contacts in our test database.

- Why you don't cover add, edit, validation and delete comment functionality.

File: cypress.json (hunk 1 above)
Comment (koropalov) [reply cypress.json:8]: Fixed

File: cypress/integration/contact/contact.test.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    cy.get('[data-cy=goBackButton]')
+      .should('be.visible')
+      .should('have.text', 'Назад')
+      .click()
+      .wait(2000);
+    cy.location('pathname', { timeout: 1000 }).should('eq', '/contacts');
+  });
+  it('The data of new contact should be add', () => {
+    cy.get('[data-cy=add-contact]').click();
+    cy.get('[data-cy=ukCartImage]').attachFile('images/test1.png');
```
Comment (moran711) [inline cypress/integration/contact/contact.test.js:64]: You can upload image like this. 
image

File: cypress/integration/contact/contact.test.js (hunk 3 above)
Comment (moran711) [reply cypress/integration/contact/contact.test.js:1]: Please change naming of values data-cy attibutes to kebab case.

File: cypress/integration/contact/contact.test.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+  uaAddress,
+  enAddress,
+  contactData
+} from './contact.variables';
+
+describe('Contacts test', () => {
+  beforeEach(() => {
+    cy.login(Cypress.env('ADMIN_LOGIN'), Cypress.env('ADMIN_PASSWORD'));
+    cy.visit('/contacts');
+    cy.get(`[href="${'/contacts'}"]`).click();
```
Comment (moran711) [inline cypress/integration/contact/contact.test.js:19]: You can use path from our configs.

File: cypress/integration/contact/contact.test.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    cy.get('[data-cy=ua-address]').children().children().first().clear();
+    cy.get('[data-cy=ua-address]').type(contactData[1].addressUa);
+    cy.get('[data-cy=en-address]').children().children().first().clear();
+    cy.get('[data-cy=en-address]').type(contactData[1].addressEn);
+    cy.get('[ data-cy=email]').children().children().first().clear();
+    cy.get('[ data-cy=email]').type(contactData[1].email);
+    cy.get('[ data-cy = save]').click();
+    cy.wait(3000);
+    cy.get('.MuiAlert-message').should('be.visible');
+    cy.get('.MuiAlert-message').contains('Успішно змінено!');
```
Comment (moran711) [inline cypress/integration/contact/contact.test.js:109]: You can get alert message from configs

File: cypress/integration/contact/contact.test.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    });
+    cy.get('[data-cy=go-back-button]').click();
+    cy.wait(2000);
+  });
+  it('The user should be deleted', () => {
+    cy.get('@table').eq(7).children().next().click();
+    cy.get('[data-cy=dialog-confirm]').last().click();
+    cy.wait(1000);
+    cy.get('.MuiAlert-message').should('be.visible');
+    cy.get('.MuiAlert-message').contains('Успішно видалено!');
```
Comment (moran711) [inline cypress/integration/contact/contact.test.js:128]: You can get alert message from configs

File: cypress/integration/contact/contact.test.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+      .should('be.visible')
+      .should('have.text', 'Назад')
+      .click()
+      .wait(2000);
+    cy.location('pathname', { timeout: 1000 }).should('eq', '/contacts');
+  });
+  it('The data of new contact should be add', () => {
+    cy.get('[data-cy=add-contact]').click();
+    cy.fixture('HORONDI.png').then((fileContent) => {
+      cy.get('[data-cy=upload-uaPhoto]').attachFile({
```
Comment (moran711) [inline cypress/integration/contact/contact.test.js:64]: Please change naming of `upload-uaPhoto` to kebab case

**Review by moran711 — APPROVED** [review]

**Review by vasylshpak — APPROVED** [review]

**Review by VitalikVoicix — APPROVED** [review]

File: cypress.json [hunk 9]
```diff
@@ -4,5 +4,5 @@
   "viewportWidth": 1920,
   "experimentalNetworkStubbing": true,
   "chromeWebSecurity": false,
-  "ignoreTestFiles": "*.variables.js"
+  "ignoreTestFiles": "*.variables.js",
```
Comment (moran711) [inline cypress.json:7]: Please remove coma at the end of this line.

File: cypress/integration/contact/contact.test.js [hunk 10]
```diff
@@ -1,65 +1,144 @@
...
+      .children()
+      .next()
+      .children()
+      .eq(0)
+      .should('be.visible');
+    cy.get('[data-cy=go-back-button]').should('be.visible').click().wait(2000);
     cy.location('pathname', { timeout: 1000 }).should('eq', '/contacts');
   });
+  it('The data of new contact should be add', () => {
+    cy.stubRequest('addContacts', addContacts).as('addContacts');
```
Comment (moran711) [inline cypress/integration/contact/contact.test.js:60]: We have only request for `addContact`.

File: cypress/integration/contact/contact.test.js [hunk 11]
```diff
@@ -1,65 +1,144 @@
...
+    cy.get('[data-cy=en-schedule]').type(contactToAdd[0].scheduleEn);
+    cy.get('[data-cy=ua-address]').type(contactToAdd[0].addressUa);
+    cy.get('[data-cy=en-address]').type(contactToAdd[0].addressEn);
+    cy.get('[ data-cy=email]').type(contactToAdd[0].email);
+    cy.get('[ data-cy=save]').click();
+    cy.wait(3000);
+    cy.get('.MuiAlert-message').should('be.visible');
+  });
+  it('Tne data should be changed', () => {
+    cy.stubRequest('updateContact', updateContact).as('updateContact');
```
Comment (moran711) [inline cypress/integration/contact/contact.test.js:89]: We also need to stub request `getContactById`. When we click on click on edit button we sent this request for server.

File: cypress/integration/contact/contact.test.js [hunk 12]
```diff
@@ -1,65 +1,144 @@
...
+        fileContent: fileContent.toString(),
+        mimeType: 'image/png',
+        filePath: '../fixtures/link.png'
+      });
+    });
+    cy.get('[data-cy=go-back-button]').click();
+    cy.wait(2000);
+  });
+  it('The user should be deleted', () => {
+    cy.stubRequest('deleteContacts', deleteContacts).as('deleteContacts');
```
Comment (moran711) [inline cypress/integration/contact/contact.test.js:138]: We have `deleteContact` request.

**Comment by vasylshpak** [comment]
please fix conflict

**Pull request merged**